### PR TITLE
report(redesign): respect prefers-color-scheme

### PR DESCRIPTION
**Pull request by Hoten**
You can test this in FF 67 beta. Just need to change OS to dark mode (for osx, Appearance -> Dark)

### CODE REVIEW

**Comment by brendankenny** [comment]
> force true incase already dark (ex: scores all 100)

really `report-ui-features.js` should be setting dark mode for fireworks anyways, not `report-renderer.js`. Not everything that runs the report renderer is showing *only* the report (PSI, web.dev), so it shouldn't be reaching out and adding classes to `body` anyways.

Can you move it in here? Not sure what's the best way to do it...e.g check for `.score100', maybe?

**Comment by Hoten** [comment]
done. can I remove the check for devtools too?

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 1]
```diff
@@ -99,7 +99,27 @@ class ReportUIFeatures {
     window.addEventListener('resize', this._updateStickyHeaderOnScroll);
     this._setupMetricDescriptionToggleElements();
     const topbarLogo = this._dom.find('.lh-topbar__logo', this._document);
-    topbarLogo.addEventListener('click', this._toggleDarkTheme);
+    topbarLogo.addEventListener('click', () => this._toggleDarkTheme());
```
Comment (Hoten) [inline lighthouse-core/report/html/renderer/report-ui-features.js]: ts error here ... How do I allow for undefined param?

```
An argument for 'force' was not provided.
```

Comment (patrickhulce) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: `@param {boolean} [force]`

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 2]
```diff
@@ -543,9 +563,10 @@ class ReportUIFeatures {
 
   /**
    * @private
+   * @param {boolean | undefined} force
```
Comment (patrickhulce) [inline lighthouse-core/report/html/renderer/report-ui-features.js]:
```suggestion
   * @param {boolean} [force]
```

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 3]
```diff
@@ -99,7 +99,27 @@ class ReportUIFeatures {
     window.addEventListener('resize', this._updateStickyHeaderOnScroll);
     this._setupMetricDescriptionToggleElements();
     const topbarLogo = this._dom.find('.lh-topbar__logo', this._document);
-    topbarLogo.addEventListener('click', this._toggleDarkTheme);
+    topbarLogo.addEventListener('click', () => this._toggleDarkTheme());
+
+    let turnOffTheLights = false;
+    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (patrickhulce) [inline lighthouse-core/report/html/renderer/report-ui-features.js]: sweeet, I guess unrecognized params just default to `.matches` false in other browsers?

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 4]
```diff
@@ -99,7 +99,27 @@ class ReportUIFeatures {
     window.addEventListener('resize', this._updateStickyHeaderOnScroll);
     this._setupMetricDescriptionToggleElements();
     const topbarLogo = this._dom.find('.lh-topbar__logo', this._document);
-    topbarLogo.addEventListener('click', this._toggleDarkTheme);
+    topbarLogo.addEventListener('click', () => this._toggleDarkTheme());
+
+    let turnOffTheLights = false;
```
Comment (patrickhulce) [inline lighthouse-core/report/html/renderer/report-ui-features.js]: super creative and awesome variable name :)

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: I think @brendankenny prefers `{T=}` which might have the same effect though?

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 3 above)
Comment (Hoten) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: yup

**Comment by paulirish** [comment]
failing on travis 

image

**Comment by Hoten** [comment]
@paulirish fixed

**Comment by paulirish** [comment]
@Hoten your merge added back the test that was removed https://github.com/GoogleChrome/lighthouse/pull/8842/files#diff-219d90e30aff619017455031ea860fec

**Comment by connorjclark** [comment]
FYI Chrome 76 will support `prefers-color-scheme`.

**Pull request merged**